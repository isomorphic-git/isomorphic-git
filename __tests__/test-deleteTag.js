/* eslint-env node, browser, jasmine */
const { makeFixture } = require('./__helpers__/FixtureFS.js')
// @ts-ignore
const snapshots = require('./__snapshots__/test-deleteTag.js.snap')
const registerSnapshots = require('./__helpers__/jasmine-snapshots')
const { E, deleteTag, listTags } = require('isomorphic-git')

describe('deleteTag', () => {
  beforeAll(() => {
    registerSnapshots(snapshots)
  })
  it('deletes the latest tag to HEAD', async () => {
    // Setup
    const { gitdir } = await makeFixture('test-deleteTag')
    // Test
    await deleteTag({
      gitdir,
      ref: 'latest'
    })
    const refs = await listTags({
      gitdir
    })
    expect(refs).toMatchSnapshot()
  })

  it('missing ref argument', async () => {
    // Setup
    const { dir, gitdir } = await makeFixture('test-deleteTag')
    let error = null
    // Test
    try {
      // @ts-ignore
      await deleteTag({ dir, gitdir })
    } catch (err) {
      error = err
    }
    expect(error).not.toBeNull()
    expect(error.code).toBe(E.MissingRequiredParameterError)
  })
})
