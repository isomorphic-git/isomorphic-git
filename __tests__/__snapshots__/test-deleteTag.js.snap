// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deleteTag deletes the latest tag to HEAD 1`] = `
Array [
  "prev",
]
`;
